### PR TITLE
Add list account cmd

### PR DESCRIPTION
**Pull request by katherinelc321**
This cmd takes an username and lists all the accounts assigned to it, or takes an account ID and prints the owner of it. Otherwise it prints all the accounts IDs grouped by owners

### CODE REVIEW

File: cmd/account/account-list.go
Comment (georgettica) [inline cmd/account/account-list.go:86]: Does that mean that any other user cannot set the ouid?

Also this means our personal IDs are defined in the source code.. is there a better way to extract this knowledge?

File: cmd/account/account-list.go
Comment (georgettica) [inline cmd/account/account-list.go:122]: Why use two practices for array expansion in the same fnction?


Which is the one you would like to follow repo wide and choose it

**Review by georgettica — COMMENTED** [review]
Looks good overall, tiny tiny nits then I will lgtm

Unless there is a logic flaw (didn't test it locally)

File: cmd/account/account-list.go
Comment (iamkirkbater) [inline cmd/account/account-list.go:27]: We aren't using the kubernetes CLI, so let's get rid of this dependency.

File: cmd/account/account-list.go
Comment (iamkirkbater) [inline cmd/account/account-list.go:66]: We don't have a dependency on kubernetes for this command, so let's remove this so that we don't need to run this on-cluster.

File: cmd/account/account-list.go
Comment (iamkirkbater) [reply cmd/account/account-list.go:86]: I wonder if we could do something like do a lookup of the OU based on name.  OU lookups though are rate-limited to only a SINGLE lookup at a time across the entire account, so if both user A and user B try to lookup the command at a single time then it will fail for whoever isn't currently doing the lookup.  And AWS Account Operator is almost _constantly_ looking this information up, making this fail _a lot_, especially now that there are multiple shards of that, so it would make this process basically awful.

We can try it, though, because maybe I'm confusing the OU lookup vs Account Lookup APIs and it might not be the _same_ problem.

File: cmd/account/account-list.go
Comment (iamkirkbater) [reply cmd/account/account-list.go:122]: I'm not sure I follow what you're asking in this comment.  Specifically what's happening here is that we're checking the key exists in the map before attempting to append the ID to it. Otherwise we'll get panics.  If the key doesn't exist, we initialize a new list as the value for that key.

File: cmd/account/account-list.go
Comment (iamkirkbater) [inline cmd/account/account-list.go:148]: AWS accounts are 12 characters long, this needs to be changed.

We could probably split on `/` and get the last value for some "future-proofing" in case AWS changes that in the future, but I'll leave that decision up to you.

File: cmd/account/account-list.go
Comment (iamkirkbater) [inline cmd/account/account-list.go:172]: Let's remove the `o.output` flag at this time, and then we can iterate back on this to add other output options in a different PR.

**Review by iamkirkbater — CHANGES_REQUESTED** [review]
There are a few changes that need to be made.

Specifically, we _need_ to change the string length for the account ID to 12 from 9, and we really should remove the kubernetes dependency.

Let's also remove the output formatting flag for this PR, and once this lands we can circle back and iterate that back in.

Let's also work together to get this into a `mgmt` subcommand, so that this doesn't collide with the already existing `account list` command, so it would be `account mgmt list`

File: cmd/account/mgmt/account-list.go
Comment (iamkirkbater) [inline cmd/account/mgmt/account-list.go:160]: Let's move this block above the `if o.username == ""` block, so move this up to line 91.  Let's also add a `return nil` after the for loop within this so we don't worry about building the hashmap when we already have the tag for this one account.

**Review by iamkirkbater — CHANGES_REQUESTED** [review]

**Review by iamkirkbater — APPROVED** [review]

**Comment by iamkirkbater** [comment]
@katherinelc321 and I paired this afternoon to make the required changes to get this done.

Should be all set to go.  We will be pairing up some time next week in order to refactor this a bit more and work on adding some unit tests.

File: cmd/account/mgmt/account-list.go
Comment (iamkirkbater) [inline cmd/account/mgmt/account-list.go:103]: We need to remove this line, or add the `for key, value...` loop that's on line 115 here.

File: cmd/account/mgmt/account-list.go
Comment (iamkirkbater) [inline cmd/account/mgmt/account-list.go:112]: We need to remove this line, or add the `for key, value...` loop that's on line 115 here.

**Review by iamkirkbater — CHANGES_REQUESTED** [review]
One small change, other than that it looks great and I'll merge once that's done.

While you're either removing the line or duplicating the for loop (I'll leave the decision on how to do that to you) can you squash this down to a single commit before you push up that change?

**Pull request merged**
